### PR TITLE
unix: fallback to blocking writes if reopening a tty fails

### PR DESCRIPTION
**Pull request by saghul**
In case reopening the file descriptor fails, fallback to the old
behavior where we do blocking writes, to avoid disrupting other
processes potentially using the file descriptor.

### CODE REVIEW

File: src/unix/tty.c [hunk 1]
```diff
@@ -54,10 +55,16 @@ int uv_tty_init(uv_loop_t* loop, uv_tty_t* tty, int fd, int readable) {
...
    */
   if (isatty(fd)) {
-    newfd = uv__open_cloexec("/dev/tty", O_RDWR);
+    r = uv__open_cloexec("/dev/tty", O_RDWR);
 
-    if (newfd < 0)
-      return newfd;    /* returned value is the error */
+    if (r < 0) {
+      /* fallback to using blocking writes */
+      if (!readable)
```
Comment (bnoordhuis) [inline src/unix/tty.c:62]: Why only when `!readable`?  Turning on O_NONBLOCK only for stdin still affects other processes that share that file descriptor (which is probably mapped to stdout and stderr as well.)

**Comment by indutny** [comment]
LGTM, but please run tests!

File: src/unix/tty.c (hunk 1 above)
Comment (indutny) [inline src/unix/tty.c:62]: That's how it was before that `dup` commit.

Comment (saghul) [inline src/unix/tty.c:62]: > Why only when !readable? Turning on O_NONBLOCK only for stdin still affects other processes that share that file descriptor (which is probably mapped to stdout and stderr as well.)

It used to be that way before, so I fallback to that. If we can safely do it for readable streams as well I can change that. Maybe it was done like that for consistency with Windows?

**Comment by saghul** [comment]
> LGTM, but please run tests!

Works On My Computer (TM) :-P

On a serious note, I'm not sure how I can simulate the fallfack. Do you?

**Comment by indutny** [comment]
You could try rebuilding node.js with your changes and running that test case, provided by user in https://github.com/joyent/libuv/issues/1207

File: src/unix/tty.c (hunk 1 above)
Comment (saghul) [inline src/unix/tty.c:62]: @bnoordhuis ping :-)

**Comment by indutny** [comment]
Seems to be working fine now! Feel free to land it.

**Comment by indutny** [comment]
Oh, also it has stopped failing :)

**Comment by indutny** [comment]
I mean even without this patch, anyway the change seems to be looking good.

**Pull request merged**